src/Atmos/Model/ref_state.jl

```diff
@@ -169,3 +170,40 @@ function (profile::LinearTemperatureProfile)(
     end
     return (T, p)
 end
+
+"""
+    DecayingTemperatureProfile{F} <: TemperatureProfile
+
+A temperature profile which decays smoothly with height `z`, until has dropped by a specified temperature difference.
+
+```math
+T(z) = \\max(T_{\\text{surface}} − ΔT \\tanh(z/H_{\\text{t}})
+```
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+struct DecayingTemperatureProfile{FT} <: TemperatureProfile
+    "surface temperature (K)"
+    T_surface::FT
+    "temperature drop from surface to top of the atmosphere (K)"
+    ΔT::FT
+    "height scale over which temperature drops (m)"
+    H_t::FT
+end
+
+function (profile::DecayingTemperatureProfile)(
+    orientation::Orientation,
+    aux::Vars,
+)
+    z = altitude(orientation, aux)
+    T = profile.T_surface - profile.ΔT * tanh(z / profile.H_t)
```

Comment:
```suggestion
    T_v = profile.T_surface - profile.ΔT * tanh(z / profile.H_t)
```
(and similar changes throughout; please make it explicit that this is virtual temperature)